packages/firebase_auth/firebase_auth/example/test_driver/test_utils.dart

```diff
@@ -1,70 +1,215 @@
+
+class EmulatorOobCode {
+  @protected
+  EmulatorOobCode({
+    this.type,
+    this.email,
+    this.oobCode,
+    this.oobLink,
+  });
+
+  final EmulatorOobCodeType type;
+  final String email;
+  final String oobCode;
+  final String oobLink;
+}
+
+enum EmulatorOobCodeType {
+  emailSignIn,
+  passwordReset,
+  recoverEmail,
+  verifyEmail,
+}
 
 String generateRandomEmail({
   String prefix = '',
   String suffix = '@foo.bar',
 }) {
   var uuid = createCryptoRandomString();
   var testEmail = prefix + uuid + suffix;
-
   return testEmail;
 }
 
-// Gets a custom token from the rnfirebase api for test purposes.
-Future getCustomToken(
-    String uid, Map<String, dynamic> claims, String idToken) async {
-  try {
-    var path = 'https://api.rnfirebase.io/auth/user/$uid/custom-token';
-    var body = json.encode(claims);
-    var headers = {'authorization': 'Bearer $idToken'};
-
-    final response = await http.post(
-      Uri.parse(path),
-      headers: headers,
-      body: body,
-    );
-    if (response.statusCode == 200) {
-      // successful, parse json
-      var jsonData = json.decode(response.body);
-      return jsonData['token'];
-    } else {
-      // response wasn't successful, throw
-      throw Exception(
-        'Unexpected response from server: (${response.statusCode}) ${response.reasonPhrase}',
-      );
-    }
-  } catch (err) {
-    throw Exception(err.toString());
+/// Deletes all users from the Auth emulator.
+Future<void> emulatorClearAllUsers() async {
+  await http.delete(
+    Uri.parse(
+        'http://$testEmulatorHost:$testEmulatorPort/emulator/v1/projects/$_testFirebaseProjectId/accounts'),
+    headers: {
+      'Authorization': 'Bearer owner',
+    },
+  );
+}
+
+/// Disable a specific user by uid.
+Future<void> emulatorDisableUser(String uid) async {
+  String body = jsonEncode({'disableUser': true, 'localId': uid});
+  await http.post(
+    Uri.parse(
+        'http://$testEmulatorHost:$testEmulatorPort/identitytoolkit.googleapis.com/v1/accounts:update'),
+    headers: {
+      'Authorization': 'Bearer owner',
+      'Content-Type': 'application/json',
+      'Content-Length': '${body.length}',
+    },
+    body: body,
+  );
+}
+
+/// Retrieve a sms phone authentication code that may have been sent for a specific
+/// phone number.
+Future<String> emulatorPhoneVerificationCode(String phoneNumber) async {
+  final response = await http.get(
+    Uri.parse(
+        'http://$testEmulatorHost:$testEmulatorPort/emulator/v1/projects/$_testFirebaseProjectId/verificationCodes'),
+    headers: {
+      'Authorization': 'Bearer owner',
+    },
+  );
+  final responseBody = Map<String, dynamic>.from(jsonDecode(response.body));
+  final verificationCodes =
+      List<Map<String, dynamic>>.from(responseBody['verificationCodes']);
+  return verificationCodes.reversed.firstWhere(
+    (verificationCode) => verificationCode['phoneNumber'] == phoneNumber,
+    orElse: () => {'code': 'NOT_FOUND'},
+  )['code'];
```

Comment:
well isn't that all concise, makes me want to go and scrub the RNFB javascript version of the dinosaur coding style 🙈

Reply:
+1 for Dart 😆